dashboard/app/entities.go

```diff
@@ -544,6 +559,7 @@ type Job struct {
 	CrashID   int64
 
 	// Provided by user:
+	KernelCommit string // if empty, it will be taken from the referenced Crash
```

Comment:
KernelBranch can also contain the exact commit for the job, and it still can after this change, right?
With the addition of KernelCommit it becomes somewhat confusing: KernelBranch can contain branch or commit, and KernelCommit can contain a commit and if it does then KernelBranch contains what? which of these take precedence?

1. Do we need KernelCommit if KernelBranch can contain a commit?
2. Can we make the data model less confusing? E.g. if we receive a user testing request with a commit, then we put it into KernelCommit rather than into KernelBranch? So that KernelCommit is always a commit and KernelBranch is always a branch.
3.  Please comment in more detail which of these fields contain what/when and what takes precedence.

Reply:
Then let's at least do clearer naming and a comment that documents what field contains what/when.

Reply:
Added the `BisectFrom` code and more comments. 
Please also take a look at this new commit: https://github.com/google/syzkaller/pull/4041/commits/b6b5d42d7b4f178ba9df7d23d468f16fe0c81465

Reply:
> Added the BisectFrom code and more comments.

LGTM

> Please also take a look at this new commit:

It changes how fields are filled. Are you sure these changes OK?

Reply:
> Are you sure these changes OK?

It would be great to check it with another pair of eyes, but at least I do not see any problems with that.

For patch testing, it stays the same. For bisections
* It's anyway more correct to take kernel repo/branch from the job itself than from the manager config.
* I did not find any real uses of `KernelCommit` for job results, so whatever we think it's better be put there should be good enough.